Source/WebCore/rendering/updating/RenderTreeUpdaterViewTransition.cpp

```diff
@@ -121,17 +121,25 @@ static RenderPtr<RenderBox> createRendererIfNeeded(RenderElement& documentElemen
         return nullptr;
 
     Ref document = documentElementRenderer.document();
-    auto newStyle = RenderStyle::clone(*style);
     RenderPtr<RenderBox> renderer;
     if (pseudoId == PseudoId::ViewTransitionOld || pseudoId == PseudoId::ViewTransitionNew) {
+        const auto* capturedElement = document->activeViewTransition()->namedElements().find(name);
+        if (pseudoId == PseudoId::ViewTransitionOld && !capturedElement->oldImage)
+            return nullptr;
+        if (pseudoId == PseudoId::ViewTransitionNew && !capturedElement->newElement)
+            return nullptr;
+
+        auto newStyle = RenderStyle::clone(*style);
         RenderPtr<RenderViewTransitionCapture> rendererViewTransition = WebCore::createRenderer<RenderViewTransitionCapture>(RenderObject::Type::ViewTransitionCapture, document, WTFMove(newStyle));
 
-        if (const auto* capturedElement = document->activeViewTransition()->namedElements().find(name))
-            rendererViewTransition->setImage(pseudoId == PseudoId::ViewTransitionOld ? capturedElement->oldImage : nullptr, capturedElement->oldSize, capturedElement->oldOverflowRect);
+        if (capturedElement)
+            rendererViewTransition->setImage(pseudoId == PseudoId::ViewTransitionOld ? *capturedElement->oldImage : nullptr, capturedElement->oldSize, capturedElement->oldOverflowRect);
 
         renderer = WTFMove(rendererViewTransition);
-    } else
+    } else {
+        auto newStyle = RenderStyle::clone(*style);
```

Comment:
Not sure there is an added value of having the `newStyle` variable to move it in the next line ? Maybe we should call `RenderStyle::clone` directly as the `createRenderer` argument.